src/rust/lib/graph-editor/src/component/visualization/renderer/example/js.rs

```diff
@@ -0,0 +1,138 @@
+            bubble.setAttributeNS(null,"stroke", "black");
+            bubble.setAttributeNS(null,"fill"  , "red");
+            bubble.setAttributeNS(null,"r"     , data[2]);
+            bubble.setAttributeNS(null,"cx"    , data[0]);
+            bubble.setAttributeNS(null,"cy"    , data[1]);
+            svgElem.appendChild(bubble);
+        });
+    }
+    "#;
+
+    let fn_set_size = r#"{
+        const root    = arguments[0];
+        const width   = arguments[1][0];
+        const height  = arguments[1][1];
+        const svgElem = root.firstChild;
+        svgElem.setAttributeNS(null, "viewBox", "0 0 " + width + " " + height);
+        svgElem.setAttributeNS(null, "width"  , width);
+        svgElem.setAttributeNS(null, "height" , height);
+    }"#;
+    JsRenderer::from_functions(fn_set_data, fn_set_size)
+}
+
+/// Returns a simple bubble chart implemented in vanilla JS. Uses an object to implement the
+/// visualization logic.
+pub fn object_sample_js_bubble_chart() -> JsRenderer {
+    let fn_prototype = r#"
+    (() => {
+        const obj = new Object();
+        obj.set_data = (root, data) => {
+            const xmlns = "http://www.w3.org/2000/svg";
+            while (root.firstChild) {
+                root.removeChild(root.lastChild);
+            }
+
+            const svgElem = document.createElementNS(xmlns, "svg");
+            svgElem.setAttributeNS(null, "id"     , "vis-svg");
+            svgElem.setAttributeNS(null, "viewBox", "0 0 " + 100 + " " + 100);
+            svgElem.setAttributeNS(null, "width"  , 100);
+            svgElem.setAttributeNS(null, "height" , 100);
+            root.appendChild(svgElem);
+
+            data.forEach(data => {
+                const bubble = document.createElementNS(xmlns,"circle");
+                bubble.setAttributeNS(null,"stroke", "black");
+                bubble.setAttributeNS(null,"fill"  , "red");
+                bubble.setAttributeNS(null,"r"     , data[2]);
+                bubble.setAttributeNS(null,"cx"    , data[0]);
+                bubble.setAttributeNS(null,"cy"    , data[1]);
+                svgElem.appendChild(bubble);
+            });
+        };
+
+        obj.set_size = (root, size) => {
+            const width   = size[0];
+            const height  = size[1];
+            const svgElem = root.firstChild;
+            svgElem.setAttributeNS(null, "viewBox", "0 0 " + width + " " + height);
+            svgElem.setAttributeNS(null, "width"  , width);
+            svgElem.setAttributeNS(null, "height" , height);
+        };
+
+         return obj;
+    })()
+    "#;
+    JsRenderer::from_object(fn_prototype).unwrap()
+}
+
+
+
+/// Returns a simple bubble chart implemented in vanilla JS. uses single functions to implement the
+/// visualization.
+pub fn constructor_sample_js_bubble_chart() -> JsRenderer {
+    let fn_constructor = r#"
+        class BubbleVisualisation {
+            set_data(root, data) {
```

Comment:
Please rename it to `onDataReceived` or something similar - the reason is what wev've been talking about. This will work on incremental data! :) Plus, in JS you have to use cammel case names (I know, ugly).